### PR TITLE
ao/openal: Restore working condition with new push API

### PR DESCRIPTION
**Pull request by LAGonauta**
OpenAL stopped working after the merge of the new audio push API.

This PR restores it to a working condition.

Summary:
- Small bug in audio_write
- ao.device_buffer was not being set
- mp_pcm_state.playing was not being set

### CODE REVIEW

File: audio/out/ao_openal.c [hunk 1]
```diff
@@ -267,6 +264,7 @@ static int init(struct ao *ao)
         goto err_out;
     }
 
+    ao->device_buffer = p->num_buffers * p->num_samples / ao->channels.num;
```
Comment (wm4) [inline audio/out/ao_openal.c:267]: That doesn't look right (but not sure). device_buffer is in PCM frames (what most of the player calls "samples"), which means 1 sample for each channel. E.g. `size_in_bytes = ao_device_buffer * channels.num * 2` for s16.

File: audio/out/ao_openal.c [hunk 2]
```diff
@@ -175,15 +175,12 @@ static void uninit(struct ao *ao)
...
-    MP_FATAL(ao, "broken.\n");
-    return -1;
-
     float position[3] = {0, 0, 0};
     float direction[6] = {0, 0, -1, 0, 1, 0};
     ALCdevice *dev = NULL;
     ALCcontext *ctx = NULL;
     ALCint freq = 0;
-    ALCint attribs[] = {ALC_FREQUENCY, ao->samplerate, 0, 0};
+    ALCint attribs[] = { ALC_FREQUENCY, ao->samplerate, 0, 0 };
```
Comment (wm4) [inline audio/out/ao_openal.c:183]: Unnecessary cosmetic change that also sort of goes against most of the coding style in this software.

**Comment by wm4** [comment]
Does normal playback end work? Try "mpv audiofile.wav --gapless=no -ss -10" and wait until it's done playing.

**Comment by LAGonauta** [comment]
> Does normal playback end work? Try "mpv audiofile.wav --gapless=no -ss -10" and wait until it's done playing.

Just tried:
`./build/mpv --ao=openal --start=-10 --gapless-audio=no {media}.ogg`

Output:
```
 (+) Audio --aid=1 (vorbis 2ch 22050Hz)
AO: [openal] 44100Hz stereo 2ch float
A: 00:00:51 / 00:00:51 (100%)

Exiting... (End of file)
```

mpv exit normally and no error was printed

File: audio/out/ao_openal.c (hunk 1 above)
Comment (LAGonauta) [reply audio/out/ao_openal.c:267]: You are right, according to the documentation `num_samples` are actually frames (complete samples):
https://mpv.io/manual/master/#audio-output-drivers-openal-num-samples

File: audio/out/ao_openal.c (hunk 2 above)
Comment (LAGonauta) [reply audio/out/ao_openal.c]: Done

File: audio/out/ao_openal.c (hunk 1 above)
Comment (LAGonauta) [reply audio/out/ao_openal.c]: Done.

**Comment by wm4** [comment]
That looks good. Can you adjust the commit subject line to project conventions? Then I'll merge.

**Comment by LAGonauta** [comment]
Hopefully correctly adjusted the commit subject line

**Comment by wm4** [comment]
Not quite. Adjusted and merged manually.

**Pull request closed without merging**